### ISSUE TITLE
Bug: Helm install missing POD_NAMESPACE on deployment, CRDs don't work

### ISSUE THREAD

**Issue by kyschouv** [Azure/azure-service-operator#1327]
**Describe the bug**
We're installing using the Helm chart, and there are a couple of issues. First of all, pods for the `azureoperator-controller-manager` deployments log an error from the `manager` container complaining that the `POD_NAMESPACE` environment variable isn't set. If the deployment is manually patched, then the pod eventually outputs this error:

```
ERROR   setup   problem running manager {"error": "no matches for kind \"MySQLAADUser\" in version \"azure.microsoft.com/v1alpha1\""}

github.com/go-logr/zapr.(*zapLogger).Error
        /go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
main.main
        /workspace/main.go:980
runtime.main
        /usr/local/go/src/runtime/proc.go:203
```

In both cases, ASO is unusable.

**To Reproduce**

Install chart version 0.1.0 using the following values file (we're using Kubernetes version 1.18.10):

Values file:
```
azureSubscriptionID: {subscription}
azureTenantID: {tenant}
azureUseMI: true
```

**Additional context**
We're using Terraform to manage this deployment, but that shouldn't impact anything, as it's still just using Helm under the hood.

Is the Helm chart still the recommended install method? Should we be using Operator Lifecycle Manager or something to manage the install? Will that work with a managed identity? (we're trying to use managed identities wherever possible).

**Comment by babbageclunk** [comment]
Thanks @kyschouv - I should have released an update for the helm chart when I did the last release, it looks like that will be the cause of both the POD_NAMESPACE and MySQLAADUser errors. Sorry about that - I'll update it today!

Deploying with Helm is still the recommended install method, although there are some issues with CRD updates for Helm that are still open - discussion here: https://github.com/helm/helm/issues/5871
We're not sure how best to handle that problem.

Installing with OLM would also work, but the latest version isn't on operatorhub yet either - I'll start that process too. (It would preventthe issue of the chart and image being out of sync.) OLM can support managed identity.

**Comment by kyschouv** [comment]
Do you have an indication of whether Helm or OLM will be a more stable method of installation for the time being (i.e. next 3-6 months)? It's feeling like OLM is probably the way to go, especially as CRDs are likely to be added and potentially changed, and that Helm issue has been open for nearly a year and a half now without resolution. However, if it's not being updated as frequently, that might point at Helm being the best choice? And just dealing with CRD challenges?

**Comment by babbageclunk** [comment]
That's a good question - I think in the long term you're right, OLM seems like it has a better story for CRD updates.  
The operator bundle is pretty new so we haven't got the update process fully worked out yet (this will be the first update for it). At the moment it's served from operatorhub so updating it means getting a PR to their community operators repo approved and merged. So it might lag behind the helm chart, since that's hosted in our repo.

**Comment by matthchr** [comment]
For those of you having this issue still, the current state of things is:
1. Helm chart version 0.1.0 - **broken**
2. Helm chart version 1.0.0 - working

The previous version of the helm chart is still using `:latest` for the reference to the container image to run, which is wrong and is the source of this problem (it's using the new code but the old chart).

I will work on fixing the old chart 0.1.0 today, but we recommend you use the newer release 1.0.0 anyway and that should unblock folks.

**Comment by ChuckkNorris** [comment]
@matthchr Thanks for clarifying. The working `helm install` command is as follows - might be able to get a PR to update the docs later, but if others struggle.
```
helm upgrade --install aso https://github.com/Azure/azure-service-operator/raw/master/charts/azure-service-operator-1.0.0.tgz \
        --create-namespace \
        --namespace=azureoperator-system \
        --set azureSubscriptionID=$AZURE_SUBSCRIPTION_ID \
        --set azureTenantID=$AZURE_TENANT_ID \
        --set azureClientID=$AZURE_CLIENT_ID \
        --set azureClientSecret=$AZURE_CLIENT_SECRET \
        --set image.repository="mcr.microsoft.com/k8s/azureserviceoperator:latest"
```

**Comment by kyschouv** [comment]
Thanks for the update. We ended up using OLM for the time-being. Is that release being kept up-to-date as well?

**Comment by matthchr** [comment]
@ChuckkNorris - you don't want the `--set image.repository="mcr.microsoft.com/k8s/azureserviceoperator:latest"` at the end ideally actually, as that means you'll use a newer version of the manager container when it's published. Obviously in the past that hasn't caused issues but it _can_ and you should have control over when you update your ASO deployment, whereas if you use `:latest` it'll just pull whatever we've published most recently the next time that pod restarts.

**Comment by matthchr** [comment]
@kyschouv - The OLM release should be mostly up-to-date. Right now we don't have all of the automation in place to automatically release there when we release here on GitHub so it can lag a bit. @babbageclunk can speak more to that.

**Comment by ChuckkNorris** [comment]
@matthchr Gotcha - so we can omit setting the `image.repository` parameter all together and the chart should grab the correct image version that works with the chart?

**Comment by matthchr** [comment]
That's exactly right, the specific image version is encoded into `values.yaml` of the helm chart for each Helm release (and will continue to be going forward). That way if we make changes (like we did with `POD_NAMESPACE`) that require a chart change _and_ a code change together everything will keep working transparently without users running into issues like they did this time.

**Comment by matthchr** [comment]
This should be all fixed now, even for the older chart 0.1.0

**Comment by ChuckkNorris** [comment]
Awesome! Thanks @matthchr!
